hypervisor/arch/x86/guest/vmexit.c

```diff
@@ -271,6 +272,42 @@ static int32_t unhandled_vmexit_handler(struct acrn_vcpu *vcpu)
 	return 0;
 }
 
+static void vcpu_complete_split_lock_emulation(struct acrn_vcpu *cur_vcpu)
+{
+	struct acrn_vcpu *other;
+	uint16_t i;
+
+	if (cur_vcpu->vm->hw.created_vcpus > 1U) {
+		foreach_vcpu(i, cur_vcpu->vm, other) {
+			if (other != cur_vcpu) {
+				bitmap_clear_lock(ACRN_REQUEST_SPLIT_LOCK, &other->arch.pending_req);
+				signal_event(&other->events[VCPU_EVENT_SPLIT_LOCK]);
+			}
+		}
+
+		put_vm_lock(cur_vcpu->vm);
+	}
+}
+
+/* MTF is currently only used for split-lock emulation */
+static int32_t mtf_vmexit_handler(struct acrn_vcpu *vcpu)
+{
+	uint32_t value32;
+
+	value32 = exec_vmread32(VMX_PROC_VM_EXEC_CONTROLS);
```

Comment:
So you decided to not use a cache of this register? If this happens rarely, it is fine.

Reply:
cached read in patch "hv: add a cache register for VMX_PROC_VM_EXEC_CONTROLS"